core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -250,6 +250,9 @@ public void run() {
       loadBalancer.shutdown();
       loadBalancer = null;
```

Comment:
In `LbHelperImpl.updateBalancingState()`, the state and the picker should be ignored if `loadBalancer != lb`, so that the channel ignores any updates from a balancer that has been shutdown and discarded.